compiler/qsc_eval/src/intrinsic.rs

```diff
@@ -36,6 +37,15 @@ pub(crate) fn call(
         #[allow(clippy::cast_precision_loss)]
         "IntAsDouble" => Ok(Value::Double(arg.unwrap_int() as f64)),
         "IntAsBigInt" => Ok(Value::BigInt(BigInt::from(arg.unwrap_int()))),
+        "DoubleAsStringWithPrecision" => {
+            let [input, prec_val] = unwrap_tuple(arg);
+            let precision = usize::try_from(prec_val.unwrap_int()).expect("integer value");
```

Comment:
I think this `expect` can fail in practice, since Q# `Int` is signed and nothing prevents a user from passing a negative number to it. You'll need to check for negative values and return the appropriate error before converting to `usize` here (and add a unit test that confirms the behavior).